src/pages/main/processor/hooks/useFileTreeEvent.ts

```diff
@@ -1,92 +1,171 @@
+            );
+        },
+      );
+      removeInvalidFileNodes(...uids);
+      dispatch(setDoingFileAction(false));
 
-        const uids: TNodeUid[] = [];
-        _targetUids.map((targetUid, index) => {
-          const { name } = _uids[index];
-          uids.push(`${targetUid}/${name}`);
-        });
-        _delete(uids);
-      } else if (type === "remove") {
+      // reload the current project
+      setReloadCurrentProjectTrigger((prev) => !prev);
+    },
+    [
+      didRedo,
+      didUndo,
+      addInvalidFileNodes,
+      removeInvalidFileNodes,
+      project,
+      fileTree,
+      fileHandlers,
+    ],
+  );
+  const _rename = useCallback(
+    async ({ orgUid, newUid }: { orgUid: TNodeUid; newUid: TNodeUid }) => {
+      const node = fileTree[orgUid];
+      if (!node) return;
+      const nodeData = node.data;
+      if (nodeData.changed && !confirmAlert(FileChangeAlertMessage)) {
+        didRedo && dispatch({ type: FileTree_Event_UndoActionType });
+        didUndo && dispatch({ type: FileTree_Event_RedoActionType });
+        return;
       }
-    } else {
-      const { type, param1, param2 } = lastFileAction;
-      if (type === "create") {
-        _create(param2);
-      } else if (type === "rename") {
-        const { uid } = param1;
-        const { newName } = param2;
-        (async () => {
-          addTemporaryNodes(uid);
-          await _rename(uid, newName);
-          removeTemporaryNodes(uid);
-        })();
-      } else if (type === "cut") {
-        const _uids: { uid: TNodeUid; name: string }[] = param1;
-        const targetUids: TNodeUid[] = param2;
 
-        const uids: TNodeUid[] = _uids.map((_uid) => _uid.uid);
-        _cut(uids, targetUids);
-      } else if (type === "copy") {
-        const _uids: { uid: TNodeUid; name: string }[] = param1;
-        const targetUids: TNodeUid[] = param2;
+      const newName = getFullnameFromUid(newUid);
+      dispatch(setDoingFileAction(true));
+      addInvalidFileNodes(node.uid);
+      await callFileApi(
+        {
+          projectContext: project.context,
+          action: "rename",
+          fileTree,
+          fileHandlers,
+          uids: [node.uid],
+          parentUid: node.parentUid as TNodeUid,
+          name: newName,
+        },
+        () => {
+          LogAllow && console.error("error while renaming file system");
+        },
+        (done: boolean) => {
+          LogAllow &&
+            console.log(done ? "successfully renamed" : "not renamed");
+        },
+      );
+      removeInvalidFileNodes(node.uid);
+      dispatch(setDoingFileAction(false));
 
-        const uids: TNodeUid[] = [];
-        const names: string[] = [];
-        _uids.map((_uid) => {
-          uids.push(_uid.uid);
-          names.push(_uid.name);
-        });
-        _copy(uids, names, targetUids);
-      } else if (type === "remove") {
-      }
-    } */
+      // reload the current project
+      setReloadCurrentProjectTrigger((prev) => !prev);
```

Comment:
instead, let's have a function that internally toggles it. Let's call it 'triggerCurrentProjectReload`

Reply:
okay, fixed it.